qr_code_app - Flutter/lib/screens/scan_screen/scan_view_model.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:flutter/material.dart';
+import 'package:qr_generator/models/seed.dart';
+import 'package:qr_generator/screens/scan_screen/scan_repository.dart';
+
+class ScanViewModel {
+  bool _isLoading = false;
+
+  Future<Object?> verifyQr(
+    String? seed,
+  ) async {
+    var commentResponse;
+
+    WidgetsBinding.instance?.addPostFrameCallback((timeStamp) {
+      _isLoading = true;
+    });
+
+    try {
+      if (seed == null) throw UnknownException();
+
+      commentResponse = await ScanRepository().verifyQr(Seed(seed, ""));
```

Comment:
**question:** Are you familiar with dependency injection?

Reply:
Yes, though I must admit I did not finalize the QR code verification feature after scanning. Didn't know what database the scanned QR code would be checked against